homeassistant/components/light/hue.py

```diff
@@ -6,212 +6,79 @@
+    if discovery_info is None:
         return
 
-    # If we came here and configuring this host, mark as done
-    if host in _CONFIGURING:
-        request_id = _CONFIGURING.pop(host)
-
-        configurator = get_component('configurator')
-
-        configurator.request_done(request_id)
-
-    lights = {}
-
-    @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
-    def update_lights():
-        """ Updates the Hue light objects with latest info from the bridge. """
-        try:
-            api = bridge.get_api()
-        except socket.error:
-            # socket.error when we cannot reach Hue
-            _LOGGER.exception("Cannot reach the bridge")
-            return
-
-        api_states = api.get('lights')
-
-        if not isinstance(api_states, dict):
-            _LOGGER.error("Got unexpected result from Hue API")
-            return
-
-        new_lights = []
-
-        api_name = api.get('config').get('name')
-        if api_name == 'RaspBee-GW':
-            bridge_type = 'deconz'
-        else:
-            bridge_type = 'hue'
-
-        for light_id, info in api_states.items():
-            if light_id not in lights:
-                lights[light_id] = HueLight(int(light_id), info,
-                                            bridge, update_lights,
-                                            bridge_type=bridge_type)
-                new_lights.append(lights[light_id])
-            else:
-                lights[light_id].info = info
-
-        if new_lights:
-            add_devices_callback(new_lights)
-
-    update_lights()
-
-
-def request_configuration(host, hass, add_devices_callback, filename):
-    """ Request configuration steps from the user. """
-    configurator = get_component('configurator')
-
-    # We got an error if this method is called while we are configuring
-    if host in _CONFIGURING:
-        configurator.notify_errors(
-            _CONFIGURING[host], "Failed to register, please try again.")
-
-        return
-
-    # pylint: disable=unused-argument
-    def hue_configuration_callback(data):
-        """ Actions to do when our configuration callback is called. """
-        setup_bridge(host, hass, add_devices_callback, filename)
-
-    _CONFIGURING[host] = configurator.request_config(
-        hass, "Philips Hue", hue_configuration_callback,
-        description=("Press the button on the bridge to register Philips Hue "
-                     "with Home Assistant."),
-        description_image="/static/images/config_philips_hue.jpg",
-        submit_caption="I have pressed the button"
-    )
+    # FIXME / debuglog
+    _LOGGER.warning('Discovery info: %s', discovery_info)
+    bridge_id = discovery_info.get(ATTR_HUE_BRIDGEID)
+    for light_id in discovery_info.get(ATTR_HUE_LIGHTS):
+        # FIXME / debuglog
+        _LOGGER.warning('Found HueLamp: %s', light_id)
+        # FIXME / Rewrite this to one add_devices call
+        add_devices([HueLight(bridge_id, light_id)])
```

Comment:
`add_devices` will filter out lights that have already been added before (based on the `unique_id` property). So you can also choose to just add all lights of this bridge and have `add_devices` do it's magic.